### ISSUE TITLE
open iTerm2 tab with customized name

### ISSUE THREAD

**Issue by alexandnpu** [fitztrev/shuttle#131]
in shuttle's settings, we usually do the following configuration

		{
			"Vagrant": [
				{
					"name": "precise32",
					"cmd": "ssh vagrant@127.0.0.1 -p 2222 -i ~/.vagrant.d/insecure_private_key"
				}
			]
		}

but when a new tab is opened, its tabname is not what I expected, "precise32", it is the "cmd"

is there some way to configure shuttle to use the name, "precise32" here, to open the new tab?

Thus, I can quickly find my machine through the tabname.

**Comment by thshdw** [comment]
Hello, 

You would do this:
```
    "Vagrant": [
        {
          "cmd": "ssh vagrant@127.0.0.1 -p 2222 -i ~/.vagrant.d/insecure_private_key",
          "name": "precise32",
          "title": "precise32"
        }
      ]
    }
```
in the above ```"cmd"``` is the command that is sent to the terminal. ```name``` is what you see in the menu and ```title``` is what you see in the tab name. Currently if the title is not set then nothing is passed as the title name. I could change this if make shuttle use the ```"name"``` as ```"title"``` if title is empty. This way you could not have to manually set the ```"title"``` Let me know if this sounds good to you!

**Comment by alexandnpu** [comment]
The configuration is fine. But I added the "title" parameter to my configuration file, It did not work. Shuttle does support the "title" parameter currently, right?

**Comment by thshdw** [comment]
It's working for me in the latest version. Make sure you have in your JSON ```"iTerm_version":"stable",``` or ```"iTerm_version":"stable",``` depending your installed version.

**Comment by alexandnpu** [comment]
@thshdw ，how to get the shuttle version that I am currently using?

**Comment by thshdw** [comment]
@alexandnpu In the apple menu press the shuttle icon then click on about. You should get a dialog box that gives you the version number.

**Comment by alexandnpu** [comment]
maybe my shuttle is too old, the 'about' menu lead me to shuttle's homepage. I installed the latest version 1.2.3, my customized title appears only when the ssh try to connect to my server. when the connection succeeded, my customized title disappeared. The cmd name specified in the shuttle.json came back again.

**Comment by thshdw** [comment]
Which version of iTerm are you using? The new changes only work with the latest stable (2.1.4) or a new Nightly build.

**Comment by alexandnpu** [comment]
ITerm2 build 2.1.4

**Comment by thshdw** [comment]
In you JSON file in the settings area you have ```"iTerm_version":"stable",``` ?

**Comment by alexandnpu** [comment]
```
{
	"_comments": [
		"Valid terminals include: 'Terminal.app' or 'iTerm'",
		"Hosts will also be read from your ~/.ssh/config or /etc/ssh_config file, if available",
		"For more information on how to configure, please see http://fitztrev.github.io/shuttle/"
	],
	"terminal": "iTerm",
    "iTerm_version": "stable",
	"launch_at_login": true,
	"show_ssh_config_hosts": false,
	"ssh_config_ignore_hosts": [],
	"ssh_config_ignore_keywords": [],
	"hosts": [
		{
```

This is the beginning of my configuration.

**Comment by thshdw** [comment]
I just ssh'd to a system using iTerm and my title stays after my connection. Do you have anything in .bash_profile or .bashrc that could be changing the title?

**Comment by alexandnpu** [comment]
on my mac:

.bash_profile
```
alias ls="ls -G"
alias vim="/usr/local/Cellar/vim/7.4.865/bin/vim"
alias vimdiff="/usr/local/Cellar/vim/7.4.865/bin/vimdiff"
PS1='\[\e[0;33m\]\u\[\e[m\] \[\e[1;34m\]\W\[\e[m\] \[\e[1;32m\]\$\[\e[m\] '
eval "$(pyenv init -)"
eval "$(pyenv virtualenv-init -)"

export HOMEBREW_GITHUB_API_TOKEN=ac2e6eb11bfd26c8e71c10bf890774aee7666eed
```

.bashrc
```
alias ls="ls -G"
PS1='\[\e[1;32m\][\u@\h \W]\$\[\e[0m\] '
```

**Comment by thshdw** [comment]
I did a test on both stable and nightly. I cant duplicate the issue...

Here is the config I tried
```
{
  "_comments": [
    "Valid terminals include: 'Terminal.app' or 'iTerm'",
    "In the editor value change 'default' to 'nano', 'vi', or another terminal based editor.",
    "Hosts will also be read from your ~/.ssh/config or /etc/ssh_config file, if available",
    "For more information on how to configure, please see http://fitztrev.github.io/shuttle/"
  ],
  "editor": "default",
  "iTerm_version": "nightly",
  "launch_at_login": false,
  "show_ssh_config_hosts": true,
  "ssh_config_ignore_hosts": [  ],
  "ssh_config_ignore_keywords": [  ],
  "terminal": "iTerm",
  "hosts": [
    {
      "cmd": "ssh me@to-my-ssh-server-ip",
      "name": "My Dev Server",
      "title": "My SSH Sever"
    }
   
  ]
}
```
I press the menu button for ```My Dev Server``` iTerm launches and runs the ssh command at this point my title is ```My SSH Server (bash)``` once I ssh to my server the title changes to ```"My SSH Server (ssh)```

**Comment by thshdw** [comment]
I just released [v1.2.4.](https://github.com/fitztrev/shuttle/releases/tag/1.2.4) In this version you don't have to set a title. If title is empty it will use the name as the title.

**Comment by unslur** [comment]
I find this question.you should set iterm>prefences>profile>terminal  disaled "Terminal may set tab/window title"
